webapp/channels/src/components/post/post_component.tsx

```diff
@@ -577,6 +588,7 @@ const PostComponent = (props: Props): JSX.Element => {
                             {!props.isPostBeingEdited &&
                             <PostOptions
                                 {...props}
+                                teamId={props.currentTeam.id}
```

Comment:
should this be `props.team.id`?

Reply:
The team.id is taken from the channel team_id, and for the reactions (which is what this is passed to) we need the current team as the team.id might not always be present.

Reply:
AFAICU, `channel.team_id` would be empty in the cases of GM/DM channels which are "teamless". In those cases it doesn't matter in which team you are so we pass the current one. But for all other channels we need to pass the correct team.

Reply:
@koox00 i removed the teamId prop due to repetition since we are taking the team. Also in the case of post options this needs to take the currentTeam ID because its only been used for the recent reactions. 

BTW if we only pass the `props.team.id` it fails the check ```caught (in promise) TypeError: Cannot read properties of undefined (reading 'id')
  ```

Reply:
@nevyangelova I understand but issue still stands, `currentTeam` !== `post.channel.team`, 
those two won't always be the same since we display posts from other teams as well as the current team.

We need to pass the post's correct team to any child component that uses team id.

The cases where a post doesn't have a team id associated with are DM/GMs that's why we use `currentTeam.id`.